### PR TITLE
OTel Semantics: Some minor changes here & there

### PR DESCRIPTION
**Pull request by rish691**
Following changes has been done:
* Use first class field for http url, method in ServiceCallViewGenerator
* Do not use otel key db.connection_string for db url
* Add OTel status code in ApiStatusEnricher
* Add OTel http host key, in the list of HOST_HEADERS in ApiBoundaryType enricher

### CODE REVIEW

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java [hunk 1]
```diff
@@ -170,25 +170,15 @@ public static boolean isSqlTypeBackendForOtelFormat(
    * @return backend uri based on otel format
    */
   public static Optional<String> getBackendURIForOtelFormat(Event event) {
-    if (SpanAttributeUtils.containsAttributeKey(event, OTEL_DB_CONNECTION_STRING)
-        && SpanSemanticConventionUtils.isValidUri(SpanAttributeUtils.getStringAttribute(event, OTEL_DB_CONNECTION_STRING))) {
-      return Optional.of(SpanAttributeUtils.getStringAttribute(event, OTEL_DB_CONNECTION_STRING));
-    } else {
-      return SpanSemanticConventionUtils.getURIForOtelFormat(event);
-    }
+    return SpanSemanticConventionUtils.getURIForOtelFormat(event);
```
Comment (kotharironak) [inline semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java:173]: Why do we need two variants of this method? - `getBackendURIForOtelFormat(Event event)` and `getBackendURIForOtelFormat(Map<String, AttributeValue> attributeValueMap)`

If, we still need,
```
public static Optional<String> getBackendURIForOtelFormat(Event event) {
  return getBackendURIForOtelFormat(event.getAtttibuteMap())
}
```

File: span-normalizer/span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/fieldgenerators/SqlFieldsGeneratorTest.java [hunk 2]
```diff
@@ -78,13 +79,18 @@ public void testPopulateOtherFields() {
...
-    map.put(OTelDbSemanticConventions.DB_SYSTEM.getValue(),
-        AttributeValue.newBuilder().setValue(OTelDbSemanticConventions.MYSQL_DB_SYSTEM_VALUE.getValue()).build());
-    map.put(OTelDbSemanticConventions.DB_CONNECTION_STRING.getValue(),
-        AttributeValue.newBuilder().setValue("jdbc:mysql://mysql:3306/shop").build());
+
+    Map<String, AttributeValue> map = Map.of(
+        OTelDbSemanticConventions.DB_SYSTEM.getValue(),
+        AttributeValue.newBuilder().setValue(OTelDbSemanticConventions.MYSQL_DB_SYSTEM_VALUE.getValue()).build(),
+        OTelDbSemanticConventions.DB_CONNECTION_STRING.getValue(),
+        AttributeValue.newBuilder().setValue("jdbc:mysql://mysql:3306/shop").build(),
```
Comment (kotharironak) [inline span-normalizer/span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/fieldgenerators/SqlFieldsGeneratorTest.java:87]: For value of `db.connection_string`, can you replace this with  - `Server=shopdb.example.com;Database=ShopDb;Uid=billing_user;TableCache=true;UseCompression=True;MinimumPoolSize=10;MaximumPoolSize=50;` ?
Ref : https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md#mysql

Comment (kotharironak) [inline span-normalizer/span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/fieldgenerators/SqlFieldsGeneratorTest.java:87]:
```suggestion
        AttributeValue.newBuilder().setValue("Server=shopdb.example.com;Database=ShopDb;Uid=billing_user;TableCache=true;UseCompression=True;MinimumPoolSize=10;MaximumPoolSize=50;").build(),
```

File: hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/ServiceCallViewGenerator.java [hunk 3]
```diff
@@ -320,13 +322,11 @@ private void buildEntrySpanView(
 
     // If this is a HTTP request, parse the http response status code.
     if (protocol == Protocol.PROTOCOL_HTTP || protocol == Protocol.PROTOCOL_HTTPS) {
+      Optional<Request> httpRequest = Optional.ofNullable(event.getHttp())
```
Comment (kotharironak) [inline hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/ServiceCallViewGenerator.java:325]: Note: I think, we should add tests for all views too (not a blocker for this PR)

**Review by kotharironak — APPROVED** [review]
lgtm

File: semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java (hunk 1 above)
Comment (rish691) [reply semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java:173]: I also wanted to make it uniform, but I think we should still rely on SpanAttributeUtils for reading from event. 
In existing place where we already read from Map, this method is useful.

File: span-normalizer/span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/fieldgenerators/SqlFieldsGeneratorTest.java (hunk 2 above)
Comment (rish691) [reply span-normalizer/span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/fieldgenerators/SqlFieldsGeneratorTest.java]: done

**Review by kotharironak — APPROVED** [review]

**Pull request merged**